### PR TITLE
Add unittest and scope stack to resolve scope issues in parsing

### PR DESCRIPTION
**Pull request by ahesselgrave**

### CODE REVIEW

**Comment by ahesselgrave** [comment]
Note that the whole file looks like it was modified because I changed from 2 spaces to 4 spaces for readability.

File: config_parser_test.cc [hunk 1]
```diff
@@ -1,11 +1,71 @@
...
 #include "gtest/gtest.h"
 #include "config_parser.h"
 
-TEST(NginxConfigParserTest, SimpleConfig) {
-  NginxConfigParser parser;
-  NginxConfig out_config;
 
-  bool success = parser.Parse("example_config", &out_config);
+// Test fixture for testing parsing a string literal
+class NginxParseStringConfigTest : public ::testing::Test {
```
Comment (kevinwu127) [inline config_parser_test.cc]: Consider merging this test fixture with the other one for better organization and less repeated code.

File: config_parser_test.cc [hunk 2]
```diff
@@ -1,11 +1,71 @@
...
+			"\tlocation / { \n"
+				"\t\tproxy_pass http://localhost:8080; \n"
+			"\t}\n"
+		"}";
+
+	EXPECT_TRUE(parse_string(config_string))
+		<< config_string;
+}
+
+TEST_F(NginxParseStringConfigTest, EmptyCurlyBlock) {
```
Comment (kevinwu127) [inline config_parser_test.cc]: Should we come to a verdict on whether or not this should be true or false?

**Review by kevinwu127 — CHANGES_REQUESTED** [review]

File: config_parser_test.cc (hunk 2 above)
Comment (ahesselgrave) [reply config_parser_test.cc]: nginx does allow empty location {} blocks, and unless we have a case-by-case situation or the instructors say something different I'm going to follow what nginx goes with. Plus it gives us a test case to enforce whatever the decision is in the future.

File: config_parser_test.cc (hunk 1 above)
Comment (ahesselgrave) [reply config_parser_test.cc]: Good idea, I forgot it was overloaded. Updating now.

**Comment by ahesselgrave** [comment]
Updated with your review suggestions. Look good?

**Pull request merged**